src/components/ScrollGradient/ScrollGradient.js

```diff
@@ -96,7 +97,10 @@ class ScrollGradient extends Component {
     }
 
     if (isClient()) {
-      window.addEventListener('resize', this.updateHandler);
+      // https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver
+      this.resizeObserver = new ResizeObserver(this.updateHandler);
```

Comment:
This is very cool 👏 

One concern I have is about browser support though. I'm looking at the support table on CanIUse and the table is not as green as I'd hoped: https://caniuse.com/#feat=resizeobserver
(For example, Firefox as v68 / July 2019... and Safari as of v13 / Sept 2019 -- that seems very close, < 1 year. 😰)

Reply:
Good catch! I've added a polyfill here to better support this.

Reply:
I realize that polyfill gets downloaded a lot but did you take a look at the build history / repo? Looks like the build is failing and there hasn't been an update in 2 years. Are you concerned about support for this new dependency?

Reply:
🤷  I've been testing with a newer implementation (https://github.com/juggle/resize-observer) and so far this has been throwing errors in our testing suite as it also requires a dependency on `MutationObserver` polyfills (https://github.com/juggle/resize-observer/issues/64#issuecomment-636805189), because I believe JSDOM doesn't mock these either